### PR TITLE
Tp 20200318 st luke template

### PR DESCRIPTION
**Pull request by jacelee2000**
---
name: St Luke certificte
about: to add new St Luke new template
  the OpenCerts repository
title: "TP - create St Luke new template"
labels: St Luke new template
assignees: ''

---

# Pull Request Guidelines for Adding Certificate Templates
This document is a work in progress but here are some basic checks. As these are only basic guidelines, meeting the below doesn't indicate there will be no issues with your pull request.

### Pre-merge checks

- [ ] Did not modify any files outside of your organisation's template folder (e.g package-lock.json or anything else)
- [ ] Ensure that your code has been [rebased](https://www.digitalocean.com/community/tutorials/how-to-rebase-and-update-a-pull-request) on top of latest OpenCerts master
- [ ] Linter issues resolved (Run `npm run lint:fix` to see issues)
- [ ] `npm run test` passes
- [ ] `npm run test:integration` passes
- [ ] [Travis Build passes](https://docs.travis-ci.com/user/for-beginners/)

### Certificate Template 
- [ ] No more than 5 templates or 25 added/modified files in the pull request
- [ ] Ensure that your .opencert file's data complies with the intentions of the OpenCerts' schema - e.g recipient related information is inside the `recipient` object, etc.
- [ ] Integration test for each template that checks that the correct rendering is done given a sample certificate
- [ ] Sample certificate file included for each template, located alongside the integration test for each template
- [ ] Sample certificates must obviously be a sample certificate
  - [ ] Obviously fictitious name
  - [ ] Obviously sample signatory images
- [ ] No fixed-size raster images as part of certificate layout
- [ ] Mobile responsive design
- [ ] Date parsing should be localised to template author's timezone
- [ ] Webpack chunking code is correct
  - [ ] Has chunking code
  - [ ] Same chunking code as the other certificates belonging to that institute
- [ ] Certificate Store Addresses have been updated
  - [ ] Template Whitelist
  - [ ] Registry
- [ ] Template should not be using resources(images etc.) on the website outside of their own folder (e.g institute logo shouldn't be used from /static because there's no guarantee it will not change)

### CODE REVIEW

**Comment by Nebulis** [comment]
There are conflicts, could you fix first ?

**Comment by jacelee2000** [comment]
Hi,
I am new to git. It could be I didn't get the latest code probably. 
Is it possible that you ignore the change for the branch "Tp 20200318 st luke template"? I will check in another set of code.

**Comment by jacelee2000** [comment]
Hi,
I managed to resolved the conflict and commited. Please try to merge again. Thank you.

**Comment by Nebulis** [comment]
There is an error while building => https://app.netlify.com/sites/vigilant-newton-0ec838/deploys/5e784162163f5b000759641e

You can run npm run deploy locally

**Comment by jacelee2000** [comment]
I use my old code, it work fine. Then I tried download new code from github, it seems there is missing folder -> "node_modules". The error is due to there is no node_modules files. Is the latest file check in previously working fine?

**Comment by Nebulis** [comment]
run `npm i` to install node modules

**Comment by jacelee2000** [comment]
The error seems is due to ./common/approvedAddresses file. I git push to github, then from github I found that the file is not require, so I delete it in github -> File Changed -> View -> Delete file.
Is there any way to revert the approvedAddresses file only?

**Comment by Nebulis** [comment]
https://stackoverflow.com/questions/215718/how-can-i-reset-or-revert-a-file-to-a-specific-revision

This or you can revert manually (open the file at the revision you want and copy paste)

**Comment by jacelee2000** [comment]
I just did push force to opencert exclude the approveaddress file. But there is change of the file changed in the github. Can you try merge again?

**Comment by jacelee2000** [comment]
If still can't can I just delete the approveaddress file directly in the github -> File Changed -> View -> Delete file? Will the file deleted only for this file changed, but not in the original repository?

**Comment by Nebulis** [comment]
I think it's ok. (According to GitHub app)

Whatever you do on the forked repo does not apply to the original repo unless you open a pull request and it's merged (same across branches)

**Comment by jacelee2000** [comment]
I just install github desktop and accidentally click fetch github-desktop, it suddenly show 237 file changed. I tried to revert the changed in github-desktop again, but the file changed still show 237. Please advise what should I do. My original change file is only 6 files exclude the approveaddress file.

**Comment by jacelee2000** [comment]
I am getting lost. Can we close this thread? I will check in delta file to another new branch. Is it possible?

**Comment by jacelee2000** [comment]
I created "TP-StLuke-20200324" new branch. Please merge & compile using the new branch. Let me know once it is done. I will delete the old branch "TP-20200318-StLukeTemplate ".

File: .eslintrc [hunk 1]
```diff
@@ -42,7 +42,6 @@
     "chai-expect/no-inner-compare": "error",
     "chai-expect/missing-assertion": "error",
     "chai-expect/terminating-properties": "error",
-    "prefer-destructuring": "off",
```
Comment (Nebulis) [inline .eslintrc]: revert this

**Comment by Nebulis** [comment]
I closed this one but I let you create the new pull request

**Pull request closed without merging**